### PR TITLE
Add GitHub workflow for release

### PR DESCRIPTION
**Pull request by phanecak-maptiler**
This PR contains `release.yml` & co. copied from Planetiler and adjusted so that later on we can use it to create `planetiler-openmaptiles.jar` for new releases.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      with:
+        github-token: ${{ github.token }}
+        script: |
+          github.rest.git.createRef({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            ref: "refs/tags/v${{ github.event.inputs.version }}",
+            sha: context.sha
+          })
+    - run: mv planetiler-dist/target/*with-deps.jar planetiler-openmaptiles.jar
```
Comment (msbarry) [inline .github/workflows/release.yml:57]: I think this should just be `mv target/*with-deps.jar ...`

File: scripts/build-release.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+set -eu
+
+./mvnw -B -ntp install jib:dockerBuild -Pflatten
```
Comment (msbarry) [inline scripts/build-release.sh:5]: you can remove `-Pflatten` - that was something specific for the multi-module planetiler repo.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    - run: sha256sum planetiler-openmaptiles.jar > planetiler-openmaptiles.jar.sha256
+    - run: md5sum planetiler-openmaptiles.jar > planetiler-openmaptiles.jar.md5
+    - name: Install GPG Private Key
+      run: |
+        echo -n "${{ secrets.OSSRH_GPG_SECRET_KEY }}" | base64 --decode | gpg --batch --import
```
Comment (msbarry) [inline .github/workflows/release.yml:62]: You can skip this - it's for publishing to maven central

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - run: ./scripts/push-release.sh ${{ github.event.inputs.version }}
+      env:
+        GITHUB_ACTOR: ${{ github.actor }}
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        IMAGE_TAGS: ${{ github.event.inputs.image_tags }}
+        MAVEN_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+        MAVEN_PASSWORD: ${{ secrets.OSSRH_PASSWORD }}
+        OSSRH_GPG_SECRET_KEY_PASSWORD: ${{ secrets.OSSRH_GPG_SECRET_KEY_PASSWORD }}
```
Comment (msbarry) [inline .github/workflows/release.yml:80]: You can skip this - it's for publishing to maven central

File: .github/workflows/release.yml (hunk 1 above)
Comment (phanecak-maptiler) [reply .github/workflows/release.yml:57]: Thanks, adjusted.

File: .github/workflows/release.yml (hunk 3 above)
Comment (phanecak-maptiler) [reply .github/workflows/release.yml:62]: Removed.

File: .github/workflows/release.yml (hunk 4 above)
Comment (phanecak-maptiler) [reply .github/workflows/release.yml:80]: Removed.

File: scripts/build-release.sh (hunk 2 above)
Comment (phanecak-maptiler) [reply scripts/build-release.sh:5]: Adjusted.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    - name: Cache data/sources
+      uses: ./.github/cache-sources-action
+    - uses: actions/setup-java@v4
+      with:
+        java-version: '21'
+        distribution: 'temurin'
+        cache: 'maven'
+        server-id: ossrh
+        server-username: MAVEN_USERNAME
+        server-password: MAVEN_PASSWORD
```
Comment (msbarry) [inline .github/workflows/release.yml:38]: you shouldn't need these server settings if you aren't publishing to maven central

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+          planetiler-openmaptiles.jar*
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - run: ./scripts/push-release.sh ${{ github.event.inputs.version }}
+      env:
+        GITHUB_ACTOR: ${{ github.actor }}
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        IMAGE_TAGS: ${{ github.event.inputs.image_tags }}
+        MAVEN_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+        MAVEN_PASSWORD: ${{ secrets.OSSRH_PASSWORD }}
```
Comment (msbarry) [inline .github/workflows/release.yml:76]: in the planetiler repo push-release.sh uploads to github package registry and maven central. I think all we care about here is uploading to dockerhub, so you could replace this step with the same thing that snapshot workflow does:

https://github.com/openmaptiles/planetiler-openmaptiles/blob/461ea675a18691a928742ab4e9ef07be68513f32/.github/workflows/snapshot.yml#L20-L26

File: .github/workflows/release.yml (hunk 5 above)
Comment (phanecak-maptiler) [reply .github/workflows/release.yml]: Removed.

File: .github/workflows/release.yml (hunk 6 above)
Comment (phanecak-maptiler) [reply .github/workflows/release.yml]: Updated. And I've added also 'Upload artifact'.

**Review by msbarry — APPROVED** [review]
Looks good!

**Pull request merged**